src/citrine/builders/design_spaces.py

```diff
@@ -0,0 +1,205 @@
+try:
+    import pandas as pd
+except ImportError:  # pragma: no cover
+    raise ImportError('pandas>=0.25 is a requirement for the builders module')
+from itertools import product
+from typing import Mapping, Sequence, List
+from citrine.informatics.design_spaces import EnumeratedDesignSpace
+from citrine.informatics.descriptors import Descriptor, RealDescriptor
+
+
+def enumerate_cartesian_product(
+    design_grids: Mapping[str, Sequence],
+    descriptors: List[Descriptor],
+    name: str = 'Enumerated Cartesian product design space',
```

Comment:
I do think these static names will increase the rate at which duplicate names are written to the platform (if used).  We might want to balance it being a bit annoying to have to specify a name everywhere against that.

Reply:
Could take on `datetime` as a dependency and generate a YY-MM-DD-hh:mm stamp. I would argue in favor of doing that for all modules. Required naming is generally annoying throughout the API.